drivers/VDI.py

```diff
@@ -616,29 +630,50 @@ def configure_blocktracking(self, sr_uuid, vdi_uuid, enable):
                     self._delete_cbt_log()
                     raise xs_errors.XenError('CBTActivateFailed',
                                              opterr=str(error))
+
             else:
-                from lock import Lock
-                lock = Lock("cbtlog", str(vdi_uuid))
-                lock.acquire()
-                try:
-                    # Find parent of leaf metadata file, if any,
-                    # and nullify its successor
-                    logpath = self._get_cbt_logpath(self.uuid)
-                    parent = self._cbt_op(self.uuid,
-                                          cbtutil.get_cbt_parent, logpath)
-                    self._delete_cbt_log()
-                    parent_path = self._get_cbt_logpath(parent)
-                    if self._cbt_log_exists(parent_path):
-                        self._cbt_op(parent, cbtutil.set_cbt_child,
-                                     parent_path, uuid.UUID(int=0))
-                except Exception as error:
-                    raise xs_errors.XenError('CBTDeactivateFailed', str(error))
-                finally:
-                    lock.release()
-                    lock.cleanup("cbtlog", str(vdi_uuid))
+                self._disable_cbt(vdi_uuid)
+
         finally:
             blktap2.VDI.tap_unpause(self.session, sr_uuid, vdi_uuid)
 
+    def _disable_cbt(self, vdi_uuid, visited=None):
+        # Prevent infite loop
```

Comment:
```suggestion
        # Prevent infinite loop
```

I don't think you need to do this, you should call for the VDI and then the same function doing the disable on the snapshots.
The snapshot won't have snapshots.

Reply:
i removed it 
indeed this protection was useless